packages/salesforcedx-vscode-soql/package.json

```diff
@@ -19,28 +19,45 @@
     "Programming Languages"
   ],
   "dependencies": {
+    "@salesforce/sobject-metadata": "0.1.4",
     "@salesforce/soql-builder-ui": "0.0.2",
     "@salesforce/soql-language-server": "0.2.1",
     "debounce": "^1.2.0",
+    "vscode-extension-telemetry": "0.1.6",
     "vscode-languageclient": "^6.1.3"
   },
   "devDependencies": {
+    "@salesforce/salesforcedx-test-utils-vscode": "49.8.0",
+    "@salesforce/ts-sinon": "1.2.2",
+    "@types/chai": "^4.0.0",
     "@types/debounce": "^1.2.0",
+    "@types/mocha": "^5",
     "@types/node": "12.0.12",
+    "@types/sinon": "^2.3.2",
     "@types/vscode": "^1.46.0",
     "@typescript-eslint/eslint-plugin": "^3.7.1",
     "@typescript-eslint/parser": "^3.7.1",
+    "chai": "^4.0.2",
+    "cross-env": "5.2.0",
     "eslint": "6.0.1",
     "eslint-plugin-header": "^3.0.0",
     "eslint-plugin-jsdoc": "^30.0.3",
     "eslint-plugin-prettier": "^3.1.4",
+    "mocha": "^5",
+    "sinon": "^7.3.1",
     "typescript": "3.7.5"
   },
+  "extensionDependencies": [
```

Comment:
This means a user won't be able to run this extension without having `salesforcedx-vscode-core` installed. I thought we wanted the soql extension to run independent to everything else.

Reply:
Good catch. @jgellin-sf you think we can get remove this extensionDependency?

Reply:
It looks like it is needed for OrgAuthInfo and ChannelService. So if we do take this dependency then when someone installs the SOQL extension then salesforcedx-vscode-core would automatically be also installed if it is not present right? That sounds ok to me. Is there any downside to that?

Reply:
We did talk about running this extension independent to the others. On the other hand, we probably want to do things the same way in these various extensions.

The only downside to taking this dependency is that if the user only wants SOQL tooling and doesn't want all of the commands, etc. in salesforcedx-vscode-core. 

This code uses OrgAuthInfo, channelService, and telemetryService from the core extension. I can duplicate or work around these functions if I need to. OrgAuthInfo is a utility class that (maybe?) could be duplicated or moved to it's own library and imported separately. The services are singletons so they don't really work as libraries imported separately. Working around those means doing things in a different way to the rest of our extensions.

Reply:
@jgellin-sf It should be ok to take this dependency. I wouldn't do the extra work now to do any additional refactoring. We need those commands from core to provide org authentication anyway.

Reply:
> The only downside to taking this dependency is that if the user only wants SOQL tooling and doesn't want all of the commands, etc. in salesforcedx-vscode-core.

I'd add that if this extension depends on the core extension, it would add an additional time for extension activation, since the extension cannot be activated until all dependent extensions are activated. (On average my measurement indicates that core extension takes 2 seconds for activation on macOS)

Reply:
@xyc Does the activation of a one extension trigger the activation of the dependent extensions?

Would that also mean that we cannot use the SOQL extension unless there is an sfdx-project.json in the root directory, because the core ext would not be activted?

Reply:
> Does the activation of a one extension trigger the activation of the dependent extensions?

Yes, the dependent extensions will be triggered.

> Would that also mean that we cannot use the SOQL extension unless there is an sfdx-project.json in the root directory, because the core ext would not be activted?

If `activate` function of core extension returns without throwing, it would be considered as successfully activated. I think last time there were some issues activating core extension in non-sfdx project context when `push-or-deploy-on-save.enabled` setting is enabled. It's worth checking out the status now if we are going with this approach.

Reply:
We are still sorting out issues when other extensions force `salesforcedx-vscode-core` to activate outside a salesforce project. If we need the soql extension to activate outside of a salesforce project (sfdx-project.json being present) please do not add a dependency to `salesforcedx-vscode-core`

Reply:
It is ok to have the sfdx-project.json restriction for now for Beta. The CodeBuilder launch experience for a SOQL only user is not clearly defined yet, so it would be premature to optimize for that now. Even then since we control the launch we could definitely create a sfdx-project.json if needed for the user.

For desktop user this is not much of a change since the current SOQL experience requires an sfdx-project.json.

Reply:
sounds good 👍